tests/model_selection/test_hyperband.py

```diff
@@ -0,0 +1,247 @@
+import pytest
+import numpy as np
+import scipy.stats as stats
+import math
+
+from sklearn.linear_model import SGDClassifier
+
+from distributed import Client
+from distributed.utils_test import loop, cluster, gen_cluster
+import dask.array as da
+
+from dask_ml.datasets import make_classification
+from dask_ml.model_selection import HyperbandCV
+from dask_ml.wrappers import Incremental
+from dask_ml.model_selection._hyperband import _top_k
+from dask_ml.utils import ConstantFunction
+import itertools
+
+
+def test_top_k():
+    in_ = [{'score': 0, 'model': '0'},
+           {'score': 1, 'model': '1'},
+           {'score': 2, 'model': '2'},
+           {'score': 3, 'model': '3'},
+           {'score': 4, 'model': '4'}]
+    out = _top_k(in_, k=2, sort="score")
+    assert out == [{'score': 4, 'model': '4'},
+                   {'score': 3, 'model': '3'}]
+
+
+@pytest.mark.parametrize("array_type,library", [("dask.array", "dask-ml"),
+                                                ("numpy", "sklearn"),
+                                                ("numpy", "test")])
+def test_sklearn(array_type, library, loop, max_iter=27):
+    with cluster() as (s, [a, b]):
+        with Client(s['address'], loop=loop):
+            chunk_size = 100  # make dask array with one chunk
+            X, y = make_classification(n_samples=chunk_size, n_features=20,
+                                       random_state=42, chunks=chunk_size)
+            if array_type == "numpy":
+                X = X.compute()
+                y = y.compute()
+                chunk_size = X.shape[0]
+
+            kwargs = dict(tol=1e-3, penalty='elasticnet', random_state=42)
+            models = {'sklearn': SGDClassifier(**kwargs),
+                      'dask-ml': Incremental(SGDClassifier(), **kwargs),
+                      'test': ConstantFunction()}
+            sgd_params = {'alpha': np.logspace(-2, 1, num=1000),
+                          'l1_ratio': np.linspace(0, 1, num=1000),
+                          'average': [True, False]}
+            all_params = {'sklearn': sgd_params, 'dask-ml': sgd_params,
+                          'test': {'value': np.linspace(0, 1)}}
+            model = models[library]
+            params = all_params[library]
+
+            search = HyperbandCV(model, params, max_iter=max_iter,
+                                 random_state=42, asynchronous=False)
+            search.fit(X, y, classes=da.unique(y))
```

Comment:
What is the right thing to test here in terms of user API?  How is a user likely to interact with this thing after it is fit?  Do they access `search.best_` or something like that?  What expectations should they have of this model?  For example should it predict the outcome well on the test dataset?  Perhaps we should test these things as well.  I find that testing user API is a bit longer-lasting as code implementations change than testing internal things.